### ISSUE TITLE
Git tag new releases & publish .ipa+.apk in github releases

### ISSUE THREAD

**Issue by emanuelb** [hsjoberg/blixt-wallet#312]
This repo has no tags, please tag the latest+newer releases:
https://github.com/hsjoberg/blixt-wallet/tags

on google-play latest release is: "Updated April 7, 2021"
https://play.google.com/store/apps/details?id=com.blixtwallet
Please tag new releases, so each tag will point to the related commit of the release, This is needed for easier/automated reproducible-builds testing.

Also consider publishing the .ipa file & .apk file in github releases: (add the .ipa & .apk files as asset)
https://github.com/hsjoberg/blixt-wallet/releases

**Comment by hsjoberg** [comment]
Hi @emanuelb.
These are great suggestions and definitely something I want to start doing.

**Comment by hsjoberg** [comment]
> This is needed for easier/automated reproducible-builds testing.

Regarding reproducible builds, this is something I still need to figure out, as with Android apps you have a private key you sign the app with.

**Comment by emanuelb** [comment]
> Regarding reproducible builds, this is something I still need to figure out

tagged versions are needed in order to avoid looking and guessing for the correct commit used to build the released version.
building release version again and using diffoscope tool between the apks to look at the diffs is the way to test it.
Will test the next tagged version when it will be available, meanwhile see my upcoming issues found during writing Containerfile to compile this repo to android release apk.

> as with Android apps you have a private key you sign the app with.

no problem at all, the compare just ignore the relevant files in V1 signing (inside META-INF), also sigs can be copied from released version to tested version with https://github.com/obfusk/apksigcopier for comparison (the way f-droid does the verification)

**Comment by hsjoberg** [comment]
@emanuelb Thank you for the information.

I'll start using git tags and GitHub releases.

The situation with [lnd](https://github.com/lightningnetwork/lnd) .aar file is still a bit messy, because there are some out-of-tree changes currently needed. I expect this to settle down fairly soon however.

Other than that, I'll make sure that a commit with the version as commit message, with a corresponding git tag in the master branch will be the same as the release in Google Play Store.

**Comment by hsjoberg** [comment]
I'll start publishing the .aar file on https://github.com/BlixtWallet/blixt-lndmobile-releases for every Blixt release, although the goal is of course to let anyone just compile it for themselves.

**Comment by emanuelb** [comment]
> although the goal is of course to let anyone just compile a matching binary for themselves.

can you add Containerfile to the repo https://github.com/BlixtWallet/blixt-lndmobile-releases for building the included Lndmobile.aar file, also worth to add sha256 checksum in the releases for the .aar file (in description for example):
https://github.com/BlixtWallet/blixt-lndmobile-releases/releases/tag/f3504db
also more meaningful tag names if possible.

**Comment by hsjoberg** [comment]
> can you add Containerfile to the repo BlixtWallet/blixt-lndmobile-releases for building the included Lndmobile.aar file

Sure, the Dockerfile in root of this repository (https://github.com/hsjoberg/blixt-wallet/blob/master/Dockerfile) used to build APK files before on CircleCI and there has been some work on getting this to migrate it to GitHub Workflow/Actions. I don't have time for this at moment but I'll add it to the project's public Trello board, https://trello.com/b/a661iaHq/blixt-wallet.

Will look into doing the other things you mentioned as well. 🙂

**Comment by hsjoberg** [comment]
As you've noted I released `v0.2.0` together with mainnet, testnet APKs and .aar lnd binary yesterday. https://github.com/hsjoberg/blixt-wallet/releases/tag/v0.2.0

sha256sums is something I'll look into at a later time.

**Comment by emanuelb** [comment]
yes, I already started the reproducible-builds testing and published the results for v2 which currently lacking detailed diff content output (not just in what files the diff exists) and also lacking separate Containerfile to generate the .aar file.

**Comment by master-fuggi** [comment]
Is the blixt-wallet-mainnet-0.6.7.apk supposed to be used for manual installation on Android devices?

**Comment by hsjoberg** [comment]
@master-fuggi Yes, all APKs published on https://github.com/hsjoberg/blixt-wallet/releases are supposed to be used for manual APK installations.

They should be the same as Google Play installations and APKs uploaded are built from the same aab file.
